common/src/java/org/apache/hadoop/hive/common/FileUtils.java

```diff
@@ -449,25 +462,18 @@ public static void checkFileAccessWithImpersonation(final FileSystem fs,
     }
 
     // Otherwise, try user impersonation. Current user must be configured to do user impersonation.
-    UserGroupInformation proxyUser = UserGroupInformation.createProxyUser(
-        user, UserGroupInformation.getLoginUser());
-    try {
-      proxyUser.doAs(new PrivilegedExceptionAction<Object>() {
-        @Override
-        public Object run() throws Exception {
-          FileSystem fsAsUser = FileSystem.get(fs.getUri(), fs.getConf());
-          ShimLoader.getHadoopShims().checkFileAccess(fsAsUser, stat, action);
-          addChildren(fsAsUser, stat.getPath(), children);
-          return null;
-        }
-      });
-    } finally {
-      FileSystem.closeAllForUGI(proxyUser);
```

Comment:
Removing this can be dangerous. Is there any reason for removing this?

Reply:
I am removing this because we moved the FIleSystem.get() call outside the recursive call, and hence i thought we can just close once for every FileSystem.get() call

Reply:
Just scanned through the code again. Issue isn't the FileSystem.get() call, but it is getting invoked recursively and for each of the invocation it could be reestablishing a connection, which is a concern.

Thanks for the note @ramesh0201.